### PR TITLE
Fix trt output name map

### LINKED ISSUE #12534: bug in tensorrt graph analysis and tensorrt engine op
1.   We should use topological sorting to traverse the graph to ensure that the nodes in the subgraph are also topological ordered. 
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/inference/analysis/subgraph_splitter.cc#79

2. We should filter the parameters here. 
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/tensorrt_engine_op.cc#99

3. There exits loops inside the graph. For example:  a = f(a), it will triger bug when setITensor during op converter process. So we should deal with the output name mapping.

### PR DESCRIPTION
**Pull request by NHZlX**
fix #12534

### CODE REVIEW

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc [hunk 1]
```diff
@@ -87,34 +87,98 @@ void DataFlowGraphToFluidPass::AddFluidOp(Node *node) {
 }
 
 void CreateTrtEngineOp(Node *node, const DataFlowGraph &graph,
-                       const framework::proto::BlockDesc &block) {
+                       framework::proto::BlockDesc &block) {
```
Comment (Superjomn) [inline paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: `framework::proto::BlockDesc*` 如果要修改的话

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc [hunk 2]
```diff
@@ -87,34 +87,98 @@ void DataFlowGraphToFluidPass::AddFluidOp(Node *node) {
...
+                                       output_names.end());
+  desc.SetOutput("Ys", output_temp);
   desc.SetType("tensorrt_engine");
 
+  std::unordered_map<std::string, std::string> output_name_map;
+  auto subgraph_nodes = func->subgraph;
+
+  for (int index = 0; index < block.ops_size(); index++) {
+    framework::proto::OpDesc *op = block.mutable_ops(index);
+    // auto &op = block.mutable_ops(index);
```
Comment (Superjomn) [inline paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: delete unnecessary comments.

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc [hunk 3]
```diff
@@ -87,34 +87,98 @@ void DataFlowGraphToFluidPass::AddFluidOp(Node *node) {
...
+    PADDLE_ENFORCE_EQ(correspond_node->name(), op->type());
+
+    std::unordered_map<std::string, size_t> var2id;
+    for (auto *in_var : correspond_node->inlinks) {
+      var2id[in_var->name()] = in_var->id();
+    }
+    // TODO(zhaolong): add comments
+    for (int i = 0; i < op->inputs_size(); i++) {
+      framework::proto::OpDesc_Var *in_var = op->mutable_inputs(i);
+      // auto &in_var = op->mutable_inputs(i);
```
Comment (Superjomn) [inline paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: same here.

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc [hunk 4]
```diff
@@ -87,34 +87,98 @@ void DataFlowGraphToFluidPass::AddFluidOp(Node *node) {
...
+    output_names.insert(x->name());
   }
-  desc.SetOutput("Ys", io);
+
+  std::vector<std::string> output_temp(output_names.begin(),
+                                       output_names.end());
+  desc.SetOutput("Ys", output_temp);
   desc.SetType("tensorrt_engine");
 
+  std::unordered_map<std::string, std::string> output_name_map;
```
Comment (Superjomn) [inline paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: add more comments to describe the logic below, something about why need the following logic, how it works.

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc (hunk 2 above)
Comment (NHZlX) [reply paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: Done

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc (hunk 3 above)
Comment (NHZlX) [reply paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: Done

File: paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc (hunk 4 above)
Comment (NHZlX) [reply paddle/fluid/inference/analysis/data_flow_graph_to_fluid_pass.cc]: Done

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**